lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -482,14 +482,12 @@ async sub _check_contract_transaction {
                 "latest"
             );
 
-            if ($decimals) {
-                my $bg_decimals = $self->get_numeric_from_hex($decimals);
-                # decimals can be 0 is that why we check by reference
-                next unless eval { $bg_decimals->isa('Math::BigFloat') };
-                $transaction_cp->{amount} = $amount->bdiv(Math::BigInt->new(10)->bpow($bg_decimals));
-            } else {
-                $transaction_cp->{amount} = $amount;
-            }
+            my $bg_decimals = $self->get_valid_decimal_places($decimals);
+            # we still needs to check the reference here since the variable
+            # check will return false if the numeric is 0
+            next unless $bg_decimals || eval { $bg_decimals->isa('Math::BigFloat') };
```

Comment:
For `eval { $bg_decimals->isa('Math::BigFloat') }` you have the same check inside `get_valid_decimal_places` function

Reply:
Yes, I also need it here because even if it is an `Math::BigFloat` the return of checking $bg_decimals will be false if the value is 0, is that why I still need to do the reference check here.